### PR TITLE
Jb37906

### PR DESCRIPTION
**Pull request by lbt**
A little work to merge the codebases
Includes some things from waaaay back.
Any review should probably start at the most recent commits and work back to get an idea of wtf is happening

### CODE REVIEW

**Comment by keto** [comment]
LGTM.

Except that I would merge this to master, as the bundled branch seems to be old an one that is behind master.

File: bin/boss [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+RuoteKit.bind_engine(storage)
+
+RuoteKit.engine.register :clear => false do
+  # register forces 'override' to false, but postion 'over' overrides that
+  # and if matching key ('.+' for catchall) is not in list, it goes to last
+  catchall Ruote::StorageParticipant, {'position' => 'over'}
+end
+
+if $pnum == 0
```
Comment (larstiq) [inline bin/boss:119]: Should this be split in a roles = ["viewer"] around line 87, and then a role check to start the webserver?

**Comment by larstiq** [comment]
Mostly LGTM. I'm not confident about this whole Ruby thread business, wouldn't be surprised if it's not optimal but I can't tell.

File: bin/boss (hunk 1 above)
Comment (lbt) [reply bin/boss:119]: probably - I'd like to rewrite a lot of the code there
The global pnum is not to my liking

**Pull request merged**